### PR TITLE
Microsoft.Extensions.Diagnostics.HealthChecks.Common README

### PR DESCRIPTION
**Pull request by dariusclay**
null

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4680)

### CODE REVIEW

File: src/Libraries/Microsoft.Extensions.Diagnostics.HealthChecks.Common/README.md [hunk 1]
```diff
@@ -18,6 +22,9 @@ Or directly in the C# project file:
 </ItemGroup>
 ```
 
+## Usage Example
+
+...coming soon...
```
Comment (joperezr) [inline src/Libraries/Microsoft.Extensions.Diagnostics.HealthChecks.Common/README.md:27]: I'm assuming that's why this PR is marked as a draft? :)

**Review by joperezr — COMMENTED** [review]
Thanks a bunch for taking care of this! Let us know when it is ready for review.

File: src/Libraries/Microsoft.Extensions.Diagnostics.HealthChecks.Common/README.md [hunk 2]
```diff
@@ -1,6 +1,22 @@
...
+- Emits healthy status when the application is started and running
+
+Manual Health Check
+- A provider that enables manual control of the application's health
+- Emits unhealthy status when `ReportUnhealthy` is called
+- Emits healthy status when `ReportHealthy` is called
+
+Telemetry Publisher
+- A publisher which emits telemetry (logs and counters) representing the application's health
+- Can be configured to log only when unhealthy reports are received
```
Comment (dariusclay) [inline src/Libraries/Microsoft.Extensions.Diagnostics.HealthChecks.Common/README.md:17]: Is there any doc we can link to for the counters/tags? Or should we mention those explicitly here:

**dotnet.health_check.reports**

- dotnet.health_check.status

**dotnet.health_check.unhealthy_checks**

- dotnet.health_check.name
- dotnet.health_check.status

**Review by geeknoid — APPROVED** [review]

File: src/Libraries/Microsoft.Extensions.Diagnostics.HealthChecks.Common/README.md (hunk 2 above)
Comment (xakep139) [reply src/Libraries/Microsoft.Extensions.Diagnostics.HealthChecks.Common/README.md:17]: I'm not sure we need these details here 🤔

File: src/Libraries/Microsoft.Extensions.Diagnostics.HealthChecks.Common/README.md [hunk 3]
```diff
@@ -18,6 +34,109 @@ Or directly in the C# project file:
...
+public static IServiceCollection AddTelemetryHealthCheckPublisher(this IServiceCollection services, IConfigurationSection section)
+public static IServiceCollection AddTelemetryHealthCheckPublisher(this IServiceCollection services, Action<TelemetryHealthCheckPublisherOptions> configure)
+```
+
+For example:
+
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddTelemetryHealthCheckPublisher();
```
Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Diagnostics.HealthChecks.Common/README.md:132]: Do folks need to call `AddHealthcChecks()` here too?

File: src/Libraries/Microsoft.Extensions.Diagnostics.HealthChecks.Common/README.md (hunk 2 above)
Comment (xakep139) [reply src/Libraries/Microsoft.Extensions.Diagnostics.HealthChecks.Common/README.md:17]: If you really want to have this reference, we can add a link to documentation that @iliar-turdushev prepared and sent to David Pine

File: src/Libraries/Microsoft.Extensions.Diagnostics.HealthChecks.Common/README.md (hunk 3 above)
Comment (dariusclay) [reply src/Libraries/Microsoft.Extensions.Diagnostics.HealthChecks.Common/README.md:132]: Yes of course :D I will add in some filler

File: src/Libraries/Microsoft.Extensions.Diagnostics.HealthChecks.Common/README.md (hunk 2 above)
Comment (dariusclay) [reply src/Libraries/Microsoft.Extensions.Diagnostics.HealthChecks.Common/README.md:17]: Done

**Comment by dariusclay** [comment]
This is ready @joperezr

**Review by xakep139 — APPROVED** [review]

**Review by joperezr — APPROVED** [review]

**Comment by joperezr** [comment]
Thanks @dariusclay !

**Pull request merged**